app/controllers/api/account.php

```diff
@@ -1103,14 +1103,23 @@
 
         // TODO after this user needs to confirm mail again
 
-        $user = $projectDB->updateDocument(\array_merge(
+        if (!$isAnonymousUser) {
+            // Remove previous unique ID.
+            $projectDB->deleteUniqueKey(\md5($user->getArrayCopy()['$collection'].':'.'email'.'='.$user->getAttribute('email')));
+        }
+
+        $document = (\array_merge(
             $user->getArrayCopy(),
             ($isAnonymousUser ? [ 'password' => Auth::passwordHash($password) ] : []),
             [
                 'email' => $email,
                 'emailVerification' => false,
             ]
-        ));
+            ));
+
+        $user = $projectDB->updateDocument($document, ['email' => $email]);
```

Comment:
As far as I checked `updateDocument` accepts only 1 param.

Reply:
Yep, I probably missed that out as my original version added a unique param to updateDocument :sweat_smile: I'll fix it now